micro_sam/sam_annotator/annotator_2d.py

```diff
@@ -98,11 +98,11 @@ def annotator_2d(
         device=device, tile_shape=tile_shape, halo=halo
     )
 
+    napari.run()
```

Comment:
The viewer is not started on purpose if `return_viewer=True`. Your changes would not be compatible with the image series annotator. There is no reason to start the viewer here, you can just do it in the code that calls the annotator.